spec.bs

```diff
@@ -570,6 +595,27 @@ expandButton.addEventListener('click', () => {
 pipWindow.document.body.append(expandButton);
 </pre>
 
+## Return to the opener tab ## {#example-return-to-tab}
+
+The {{Window/focus()}} API can be used to focus the opener tab from a
+picture-in-picture window (requiring a user gesture):
+
+<pre class="lang-javascript">
+const addReturnToTabButtonScript = pipWindow.document.createElement('script');
+addReturnToTabButtonScript.src = 'addReturnToTabButton.js';
+pipWindow.document.body.append(addReturnToTabButtonScript);
+</pre>
+
+addReturnToTabButton.js:
```

Comment:
Why not follow the pattern from the example above instead of inserting a script tag?
This can be as simple as :

```
<pre class="lang-javascript">
const returnToTabButton = pipWindow.document.createElement('button');
returnToTabButton.textContent = 'Return to opener tab';
returnToTabButton.addEventListener('click', () => {
  opener.focus();
});
pipWindow.document.body.append(returnToTabButton);
</pre>
```

Reply:
Because IIUC, that `opener.focus()` is run in the context of the main page, so "opener" does not refer to the main page. This pattern allows the js to run in the PiP window context

Reply:
Also, if you instead to `window.focus()` there so you're referring to the correct window, then it's still run in the main page which does not have the user gesture.

I do agree that it is a bit weird and not ergonomic. Maybe there's a way we can make it work the `window.focus()` way. I'll think about it. Possibly we could just require the user gesture to be in the PiP window (and maybe the PiP window having system focus). And not care where it's being run

Reply:
Okay I looked into it and removed the incumbent realm aspect, and now it just requires that the PiP window has a user gesture (which gets consumed). This means that `window.focus()` from the opener realm works and `opener.focus()` from the pip realm works. I've updated the example to be the simpler case (opener realm). Thanks for the feedback!